libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/RouteAlternativeComponentImpl.kt

```diff
@@ -0,0 +1,18 @@
+package com.mapbox.navigation.ui.app.internal.controller
+
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.maps.internal.ui.RouteAlternativeContract
+
+class RouteAlternativeComponentImpl(private val store: Store) : RouteAlternativeContract {
+
+    override fun onAlternativeRoutesUpdated(
+        legIndex: Int,
+        mapboxNavigation: MapboxNavigation,
+        updatedRoutes: List<NavigationRoute>
+    ) {
+        store.dispatch(RoutesAction.SetRoutes(routes = updatedRoutes, legIndex = legIndex))
```

Comment:
Is it really `SetRoutes`? Alternative observer is invoked not when the alternatives routes are set but when they are available, if I'm not mistaken.

Reply:
Ah, this only _will_ set the routes, right?

Reply:
Yes, this will dispatch an action that will set routes to `mapboxNavigation`.